core/src/storage2/lazy/lazy_cell.rs

```diff
@@ -307,6 +307,37 @@ where
     pub fn get_mut(&mut self) -> Option<&mut T> {
         self.load_entry_mut().value_mut().into()
     }
+
+    /// Sets the value in this cell to `value`, without executing any reads.
+    ///
+    /// # Note
+    ///
+    /// No reads from contract storage will be executed.
+    ///
+    /// This method should be preferred over dereferencing or `get_mut`
+    /// in case the returned value is of no interest to the caller.
+    ///
+    /// # Panics
+    ///
+    /// If accessing the inner value fails.
+    pub fn set(&mut self, new_value: T) {
+        // SAFETY: This is critical because we mutably access the entry.
+        let cache = unsafe { &mut *self.cache.get_ptr().as_ptr() };
+        if cache.is_none() {
+            // Cache is empty, so we simply set the cache to the value.
+            // The key does not need to exist for this to work, we only need to
+            // write the value into the cache and are done. Writing to contract
+            // storage happens during setup/teardown of a contract.
+            *cache = Some(Entry::new(Some(new_value), EntryState::Preserved));
+        } else {
+            //  Cache is already populated we simply overwrite its already existing value.
+            cache
+                .as_mut()
+                .expect("unpopulated cache entry")
+                .put(Some(new_value));
+        }
+        debug_assert!(cache.is_some());
```

Comment:
That's a good idea! We should have more `debug_assert` statements in the whole codebase for such purposes.